src/ripple/app/tests/PayChan_test.cpp

```diff
@@ -0,0 +1,689 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2012, 2013 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <BeastConfig.h>
+#include <ripple/basics/chrono.h>
+#include <ripple/protocol/Feature.h>
+#include <ripple/protocol/Indexes.h>
+#include <ripple/protocol/JsonFields.h>
+#include <ripple/protocol/TxFlags.h>
+#include <ripple/test/jtx.h>
+
+#include <chrono>
+
+namespace ripple
+{
+namespace test
+{
+struct PayChan_test : public beast::unit_test::suite
+{
+    static
+    uint256
+    channel (ReadView const& view,
+        jtx::Account const& account,
+        jtx::Account const& dst)
+    {
+        auto const sle = view.read (keylet::account (account));
+        if (!sle)
+            return beast::zero;
+        auto const k = keylet::payChan (account, dst, (*sle)[sfSequence] - 1);
+        return k.key;
+    }
+
+    static Buffer
+    signClaimAuth (PublicKey const& pk,
+        SecretKey const& sk,
+        uint256 const& channel,
+        STAmount const& authAmt)
+    {
+        Serializer msg;
+        std::uint8_t const hdr[4] = {'C', 'L', 'M', 0};
```

Comment:
Should this be a `HashPrefix`? See: https://github.com/ripple/rippled/blob/develop/src/ripple/protocol/impl/HashPrefix.cpp